### PR TITLE
Define sizeof userfile at the right place

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: 

One-line summary:
Define sizeof userfile at the right place

Additional description (if needed):
Fixes compiler warning (3) in #703
Fixes magic number in commit 9e6f2790f10dff7f38cbfeaf30acb2e14e89b517
Add some comments

Test cases demonstrating functionality (if applicable):

### CODE REVIEW

File: src/userrec.c [hunk 1]
```diff
@@ -532,7 +532,7 @@ void sort_userlist()
 void write_userfile(int idx)
 {
   FILE *f;
-  char *new_userfile;
+  char new_userfile[sizeof userfile + 4]; /* 4 = sizeof "~new" */
```
Comment (thommey) [inline src/userrec.c]: at least the comment is misleading, 4 = strlen("~new"), 5 = sizeof("~new")

**Review by thommey — CHANGES_REQUESTED** [review]

File: src/userrec.c (hunk 1 above)
Comment (michaelortmann) [reply src/userrec.c]: ty, fixed.

File: src/main.c [hunk 2]
```diff
@@ -615,7 +614,7 @@ static void do_arg()
 
 void backup_userfile(void)
 {
-  char s[sizeof userfile + 4];
+  char s[sizeof userfile + 4]; /* 4 = sizeof "~bak" */
```
Comment (Cizzle) [inline src/main.c]: Same comment applies here.

Comment (michaelortmann) [reply src/main.c]: Done.

**Comment by thommey** [comment]
Is defining it in a headerfile really the way to go? It seems unconventional and could define the symbol in multiple translation units without them being related. I don't think this is the right general solution for the issue of all those char arrays.

**Comment by michaelortmann** [comment]
Thank you thommey, i committed a new/better solution, where i moved the function backup_userfile() from main.c to userrec.c, **ready for review again**.

**Review by thommey — APPROVED** [review]

**Pull request merged**